### ISSUE TITLE
[Issue] azd up fails on todo-csharp-cosmos-sql

### ISSUE THREAD

**Issue by timheuer** [Azure/azure-dev#1421]
**Describe the bug**
WHen performing `azd up` on the csharp cosmos sql template it fails with:
```
2023/01/19 13:55:26 command_runner.go:79: Run exec: 'dotnet publish D:\source\repos\todo-csharp\src\api -c Release --output C:\Users\timheuer\AppData\Local\Temp\azd3394537926'
  (✓) Done: Deploying service api

ERROR: deploying service: packaging service api: dotnet publish on project 'D:\source\repos\todo-csharp\src\api' failed: exit code: 1, stdout: MSBuild version 17.5.0-preview-22620-02+a6f6699d1 for .NET
  Determining projects to restore...
  All projects are up-to-date for restore.
C:\Program Files\dotnet\sdk\7.0.200-preview.22628.1\Current\SolutionFile\ImportAfter\Microsoft.NET.Sdk.Solution.targets(36,5): error NETSDK1194: The "--output" option isn't supported when building a solution. [D:\source\repos\todo-csharp\src\api\Todo.Api.sln]
, stderr: : exit status 1
```

This is calling `dotnet publish` on a directory that has both a SLN and csproj and will give preference to SLN if a proj is not specified.  `--output` isn't supported there.

**Comment by jongio** [comment]
@rajeshkamal5050 could you have a look?

**Comment by rajeshkamal5050** [comment]
@hemarina can you triage this and please do check with @timheuer on his setup/dependencies?

**Comment by timheuer** [comment]
Correct: https://github.com/dotnet/sdk/issues/15607

**Comment by rajeshkamal5050** [comment]
Yes, we should be updating templates. Provided the recommended fix is backward compatible and works with previous dotnet runtime/sdks.

**Comment by baronfel** [comment]
I can confirm that explicitly specifying the project file (not just the containing directory) works in previous dotnet SDKs.

**Comment by rajeshkamal5050** [comment]
@hemarina we should probably get this fixed definitely for our Apr release.

**Comment by rajeshkamal5050** [comment]
Marking this as blocker for 0.8 April 5th release. If not needed, please clarify and remove blocker.

@savannahostrowski @ellismg

**Comment by rajeshkamal5050** [comment]
Closed as part of #1779
